### PR TITLE
Add support for Versal ACAP VCK190

### PR DESCRIPTION
**Pull request by ldts**
System install:

The Versal ACAP user will need to download and install Petalinux and the Versal ACAP BSP package [1]

```
  $ ls
    xilinx-vck190-v2022.1-04191534.bsp

  $ mkdir -p optee-project
  $ cd optee-project
  $ repo init -u https://github.com/OP-TEE/manifest.git -m versal.xml
  $ repo sync -j4 --no-clone-bundle
  $ cd build
  $ make -j8 toolchains
  $ make -j8
```

The BSP will need to be upacked somewhere in the filesystem; to do that, once Petalinux has been installed do the following:

```
  $ cd ../../
  $  . /path/to/petalinux.2022.1/settings.sh
  $  petalinux-create --type project -s xilinx-vck190-v2022.1-04191534.bsp
  $ ls
    xilinx-vck190-2022.1
```

To build the Versal ACAP image, versal/bootImage-versal-vck190.bif needs to be manually modified so some of the firmware files can be reached from the BSP.

Once done

```
  $ cd optee-project/build
  $ make -f versal.mk image
```

At that point the images are in optee-project/build/versal

To boot from JTAG:
```
  $ cd versal
  $ ./boot_jtag.sh
```

To boot from SD card create a big enough bootable partition on the SD and copy the images
```
  $ cp BOOT.BIN  /mnt/usr/boot_partition/
  $ cp versal-vck190.ub  /mnt/usr/boot_partition/
```

Plug it in the device, set the boot switches and powr up the board

  At the u-boot shell stop, load the itb and boot it.

```
  uboot> mmc dev 0
  uboot> fatload mmc 0:1 0x20000000 versal-vck190.ub
  uboot> bootm 0x20000000
```

  At the login prompt type "root" to enter and execute xtests

```
  [    2.391905] mmc0: new high speed SDHC card at address aaaa
  [    2.397907] mmcblk0: mmc0:aaaa SS08G 7.40 GiB
  OP-TEE embedded distrib for versal-vck190
  buildroot login: root
  # xtest
  Run test suite with level=0
  TEE test application started over default TEE instance
  [...]
```

[1] https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/2347204609/2022.1+Release

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

### CODE REVIEW

**Comment by ldts** [comment]
@jforissier when should I send the PR to the manifest and the docs? after this has been reviewed

File: versal.mk [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+PLATFORM = versal-vck190
+OPTEE_OS_PLATFORM = versal
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH		?= $(ROOT)/arm-trusted-firmware
+U-BOOT_PATH		?= $(ROOT)/u-boot-xlnx
+BOOTGEN_PATH		?= $(ROOT)/bootgen
+LINUX_PATH		?= $(ROOT)/linux-xlnx
```
Comment (jforissier) [inline versal.mk:20]: Minor: does the `-xlnx` suffixes really help anything? Other platforms don't have them.

**Review by jforissier — APPROVED** [review]
One minor comment, in any case LGTM:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

And yes please do create PRs for the manifest and doc.

File: versal.mk (hunk 1 above)
Comment (ldts) [reply versal.mk:20]: yes you are right, it doesnt. will remove. let me update

Comment (ldts) [reply versal.mk]: going to retest everything and then send the manifest. 

BTW - just a side note-  the qemu target is no longer in the zynqmp.mk and the  docs refer to it 
https://optee.readthedocs.io/en/latest/building/devices/zynqmp.html
am I missing something?

**Pull request merged**